File: String/1849.Splitting-a-String-Into-Descending-Consecutive-Values/1849.Splitting-a-String-Into-Descending-Consecutive-Values.cpp
typedef long long ll;
class Solution {
public:
    bool splitString(string s) 
    {        
        for (int len=1; len < s.size(); len++)
        {
            ll num = stoll(s.substr(0, len));
            if (num >= 1e10) break;
            if (dfs(s, len, num))
                return true;
        }
        return false;
    }
    
    bool dfs(string&s, int cur, ll num)
    {
        if (cur==s.size()) return true;
        
        for (int len = 1; len + cur <= s.size(); len++)                
        {
            ll num2 = stoll(s.substr(cur, len));
            if (num2 >= 1e10) break;            
            if (num-1 == num2 && dfs(s, cur+len, num2))
                return true;                
        }
        return false;
    }
};

File: String/1849.Splitting-a-String-Into-Descending-Consecutive-Values/1849.Splitting-a-String-Into-Descending-Consecutive-Values.py
class Solution(object):
    def splitString(self, s):
        """
        :type s: str
        :rtype: bool
        """
        def dfs(s, cur, num):
            if cur==len(s):
                return True
            for l in range(1, len(s)-cur+1):
                if num-1==int(s[cur:cur+l]) and dfs(s, cur+l, num-1):
                    return True
            return False        
        
        for l in range(1, len(s)):
            if dfs(s, l, int(s[0:l])):
                return True
        return False
